### ISSUE TITLE
Remove God Mode

### ISSUE THREAD

**Issue by jayrogers92** [rajyraman/Levelup-for-Dynamics-CRM#70]
I've seen a couple of issues logged on here about god mode, one of them being "Users without System Admin role have access to God Mode" but there is no reply on it. 

It is a broad request, but can you remove God Mode altogether or code it so that it only works if the logged in user has System Admin role please? It is very dangerous having this tool accessible to every user? 

There is also discussion in our team as to setting the tool to only work for System Admins altogether? The tool is brilliant but gives basic users to many options.

**Comment by Sunithadevadass** [comment]
I totally agree with the above. God Mode is useful tool for admins but at the same extremly dangerous as it can be accessible by everyone. As suggested above, can it be made available to system admins only please?

**Comment by pshayer** [comment]
Great tool but giving to much power to non-admin users. Will be a loss for developers and System Admins if Group security decided to block it. I believe this tool should comply with Dynamics security model.

**Comment by rappen** [comment]
IT does comply to the security model. It only affects pure UI features, which the security model does not cover. Security is implemented server side from a data integrity perspective.
UI features are just that - UI features. These are easily possible to manipulate using features already in the browser.
So make sure your security models are correct! 😊

**Comment by pshayer** [comment]
Hi @rappen Thanks for your reply. I understand what you mean. Anyone with some understanding of JavaScript can easily manipulate Dynamics UI features. But we don't expect internal Dynamics users to make such effort. Here they don't need to make any effort, they just need to download chrome extension and click a button.

**Comment by jayrogers92** [comment]
I understand what you're saying, but fields that are simply read only or fields that are hidden on forms; the only way to stop users from revealing them/editing them on the form would be to apply field security to every read only field.

**Comment by rajyraman** [comment]
There was similar issue logged in the past. See #53 . While it is easy to fix this, I don't think this is going to make the form more secure, unless you validations are enforced using FLS, security roles and server side validations.

**Comment by vgarre** [comment]
There are fields that we are prevented from putting field level security on.  God Mode allows users to update those fields when they shouldn't.  How do we prevent that?  or how do we disable God Mode for a specific environment such as a production environment.

**Comment by J76786786** [comment]
Im surprised by the non understanding of @rappen  in this thread. Level up is a great tool in the hands of the right users. When users start to take the business processes in their own hands with god mode it introduces a huge work load on Support team to help out on stupid mistakes made with God mode. If D365 CE is going for global customers Level up cannot exist as is today. Maybe smaller Customers without fixed global processes like this but this will become/is already a problem for MS when trying to sell CE.

**Comment by J76786786** [comment]
Yes, thank you. That would be very appreciated! Can you email me at johanevert1@gmail.com

**Comment by Zahwy** [comment]
@vgarre would you please tell us the details as we are facing the same issue

**Comment by Zahwy** [comment]
> We found a way around it. Our security group set up something in our network that only allows specific profiles to run LevelUp in our environments.
> 
> I can get details if you are interested :)


@vgarre would you please tell us the details as we are facing the same issue

**Comment by vgarre** [comment]
Good Morning,

They have only been able to tell me they are doing it with a group policy that only allows specific profiles to use LevelUp.



From: Zahwy ***@***.***>
Sent: Wednesday, December 14, 2022 7:32 AM
To: rajyraman/Levelup-for-Dynamics-CRM ***@***.***>
Cc: Voni Garrett ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [rajyraman/Levelup-for-Dynamics-CRM] Remove God Mode (#70)

Caution: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe
- Libbey IT
________________________________

We found a way around it. Our security group set up something in our network that only allows specific profiles to run LevelUp in our environments.

I can get details if you are interested :)

@vgarre<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvgarre&data=05%7C01%7Cvgarre%40libbey.com%7C38f0c17819c7458cfacc08daddcf3d61%7C4663047b11554c6dbafa5ec38a099888%7C0%7C0%7C638066179425639932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=V1Vu5EE0HWWOa1egpqbDMEN%2Bz2mZm8LFclLTBVkHPoA%3D&reserved=0> would you please tell us the details as we are facing the same issue

-

**Comment by vgarre** [comment]
***@***.***

From: Zahwy ***@***.***>
Sent: Wednesday, December 14, 2022 7:32 AM
To: rajyraman/Levelup-for-Dynamics-CRM ***@***.***>
Cc: Voni Garrett ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [rajyraman/Levelup-for-Dynamics-CRM] Remove God Mode (#70)

Caution: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe
- Libbey IT
________________________________

We found a way around it. Our security group set up something in our network that only allows specific profiles to run LevelUp in our environments.

I can get details if you are interested :)

@vgarre<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvgarre&data=05%7C01%7Cvgarre%40libbey.com%7C38f0c17819c7458cfacc08daddcf3d61%7C4663047b11554c6dbafa5ec38a099888%7C0%7C0%7C638066179425639932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=V1Vu5EE0HWWOa1egpqbDMEN%2Bz2mZm8LFclLTBVkHPoA%3D&reserved=0> would you please tell us the details as we are facing the same issue

-

**Comment by Zahwy** [comment]
> Good Morning, They have only been able to tell me they are doing it with a group policy that only allows specific profiles to use LevelUp. From: Zahwy ***@***.***> Sent: Wednesday, December 14, 2022 7:32 AM To: rajyraman/Levelup-for-Dynamics-CRM ***@***.***> Cc: Voni Garrett ***@***.***>; Mention ***@***.***> Subject: [EXTERNAL] Re: [rajyraman/Levelup-for-Dynamics-CRM] Remove God Mode (#70) Caution: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe - Libbey IT
> […](#)
> ________________________________ We found a way around it. Our security group set up something in our network that only allows specific profiles to run LevelUp in our environments. I can get details if you are interested :) @vgarre<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvgarre&data=05%7C01%7Cvgarre%40libbey.com%7C38f0c17819c7458cfacc08daddcf3d61%7C4663047b11554c6dbafa5ec38a099888%7C0%7C0%7C638066179425639932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=V1Vu5EE0HWWOa1egpqbDMEN%2Bz2mZm8LFclLTBVkHPoA%3D&reserved=0> would you please tell us the details as we are facing the same issue - Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frajyraman%2FLevelup-for-Dynamics-CRM%2Fissues%2F70%23issuecomment-1351272952&data=05%7C01%7Cvgarre%40libbey.com%7C38f0c17819c7458cfacc08daddcf3d61%7C4663047b11554c6dbafa5ec38a099888%7C0%7C0%7C638066179425639932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WumaSdO6B3G1PTTGhilOfaLPvKXtbmJQTqKwiZbA3f0%3D&reserved=0>, or unsubscribe<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAQACLZZPNFFWVDNW3ZFRON3WNG45FANCNFSM4GFLCS2A&data=05%7C01%7Cvgarre%40libbey.com%7C38f0c17819c7458cfacc08daddcf3d61%7C4663047b11554c6dbafa5ec38a099888%7C0%7C0%7C638066179425639932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=aizn2EOT68LcFEG2imrVSGrH%2B15lQf8hnV24xnAbD2M%3D&reserved=0>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Great, Thanks @vgarre

**Comment by vgarre** [comment]
I’m getting more info from our SecOps  😊

The companies IT Admins need to install these .admx templates into their group policy.

https://support.google.com/chrome/a/answer/187202?visit_id=638066207799454278-2943215899&amp;rd=2

Installing is easy. Its like 4 or 5 simple steps.

Then
***@***.***



From: Voni Garrett
Sent: Wednesday, December 14, 2022 8:20 AM
To: rajyraman/Levelup-for-Dynamics-CRM ***@***.***>
Subject: RE: [EXTERNAL] Re: [rajyraman/Levelup-for-Dynamics-CRM] Remove God Mode (#70)

***@***.***

From: Zahwy ***@***.******@***.***>>
Sent: Wednesday, December 14, 2022 7:32 AM
To: rajyraman/Levelup-for-Dynamics-CRM ***@***.******@***.***>>
Cc: Voni Garrett ***@***.******@***.***>>; Mention ***@***.******@***.***>>
Subject: [EXTERNAL] Re: [rajyraman/Levelup-for-Dynamics-CRM] Remove God Mode (#70)

Caution: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe
- Libbey IT
________________________________

We found a way around it. Our security group set up something in our network that only allows specific profiles to run LevelUp in our environments.

I can get details if you are interested :)

@vgarre<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvgarre&data=05%7C01%7Cvgarre%40libbey.com%7C38f0c17819c7458cfacc08daddcf3d61%7C4663047b11554c6dbafa5ec38a099888%7C0%7C0%7C638066179425639932%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=V1Vu5EE0HWWOa1egpqbDMEN%2Bz2mZm8LFclLTBVkHPoA%3D&reserved=0> would you please tell us the details as we are facing the same issue
